lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -72,10 +72,19 @@ public void close() {
 	public class Channels extends HashMap<String, Channel> {
 		public Channels() {
 			/* remove all channels when the connection is closed, to avoid stalled state */
-			connection.on(ConnectionState.closed, new ConnectionStateListener() {
+			connection.on(new ConnectionStateListener() {
```

Comment:
Ah, that's interesting.  I suspect that affects a lot of libraries in fact.

I will raise a separate issue to cover that in docs.  @saiber I don't think you should worry about that though for this PR as it's a separate issue that needs addressing on its own.